password.py

```diff
@@ -1,49 +1,302 @@
+            continue
+
+    account_dict_file.close()
+    
     return
+    
+#<---- THIRD MAJOR FUNCTION ---->
+# Manages and Rates your passwords.
+def password_manager():
+
+    if password_randomizer_query() == 'no':
+        while True:
+            password = input("Enter a Password:\n")
+            # Fail Case 1
+            if len(password) < 6:
+                print("Password has to be 6 characters or more.")
+                continue
+            # Fail Case 2
+            elif any(character.isdigit() for character in password) == False:
+                print("Password has to include at least one digit/number.")
+                continue
+            # Fail Case 3
+            elif bool(re.match("^[a-zA-Z0-9_]*$", password)) == True:
+                print("Password has to include at least one special character")
+                continue
+            else:
+                account_info_dict["Password"] = password
+                password_rating(password)
+                print("Would you like to save your password? (Type 'yes' or 'no'): ")
+                save_query = input("")
+                if save_query == 'yes':
+                    password_save()
+                    print("Password Saved.")
+                    break
+                elif 'no':
+                    continue
+                else:
+                    break
+    else:
+        while True:
+            print("Would you like to save your password?(Type 'yes' or 'no'): ")
+            save_query = input("")
+            if save_query == 'yes':
+                password_save()
+                print("Password Saved.")
+                break
+            elif 'no':
+                password_randomizer_query()
+                continue
+            else:
+                break
+
+
 # Saves your passwords
 def password_save():
 
-    return
+    account_file = open('password.txt','w')
+    account_file.write(str(account_info_dict))
+    account_file.close()
+
+    return account_info_dict
+
 # Gives the password a rating
 def password_rating(string):
+    import collections
+    repeat_dict = collections.defaultdict(int)
+
+    for characters in string:
+        repeat_dict[characters] += 1
+
+    for characters in sorted(repeat_dict, key=repeat_dict.get, reverse=True):
+        if repeat_dict[characters] > len(string)//2:
+            print("*WARNING* This password is Bad!")
+            break
+        elif repeat_dict[characters] == 1:
+            print("This password is Safe!")
+            break
+        else:
+            print("This password is Okay.")
+            break
 
-    return
 # Lists the password
 def list_password():
+    account_file = open('password.txt','r')
+    print(str(account_file.read()))
+    account_file.close()
 
-    return
 # Main Function
 def main():
 
-    return
+    print("-"*29, "\nWelcome to Password Manager!"), print("-"*29)
+    print('*IMPORTANT* Type "recovery" if you want to recover your password!')
```

Comment:
I don't quite understand the implementation here. What does a password recovery mean? What is the use case here? Can you explain a little bit about how this program is supposed to be used?

Reply:
Thanks, good point.